library/modules/EventManager.cpp

```diff
@@ -183,26 +188,93 @@ static const eventManager_t eventManager[] = {
     manageInteractionEvent,
 };
 
+template<typename T>
+class event_tracker { //todo: use inheritance? stl seems to use variadics, so it's unclear how well that would actually work
+private:
+    std::unordered_map<T, int32_t> seen;
```

Comment:
A variant with a multimap `seen` might be good (thinking repeat jobs).

Reply:
I haven't specifically tried to test repeat jobs, but I don't think any changes are necessary at this point having removed the id from checks. As is completed jobs only have one tick (the tick they can be detected completed on) to be recognized. Since the method of doing this is the same as the old way, it should work if it did work. Maybe this could be useful somewhere else though..?